### PR TITLE
[124] [I3BXGG] Output is not same as hive when data contains '\0'

### LINKED ISSUE #124: Output is not same as hive when data contains '\0'
### Software Environment:
- OpenLooKeng version (source or binary): 1.1.0

- OS platform & distribution (eg., Linux Ubuntu 16.04): Linux Ubuntu 16.04

- Java version: openJDK 1.8.0_252

### Describe the current behavior
There is difference in output between openlooKeng CLI(1.1.0) and hive CLI(3.1.2).

### Describe the expected behavior
Output should be same between openlookEng and hive CLI.

### Steps to reproduce the issue
1.At hive side
create table test(aa varchar(20),bb int);
insert into test values('hive\0spark',1);
select * from test;
2. At openlooKeng side
select * from test;

### Related log/screenshots
Gitee Issue Link: https://e.gitee.com/open_lookeng/issues/list?issue=I3BXGG

### Special notes for this issue

### PR DESCRIPTION
**Pull request by aman-omer**
### What type of PR is this?

/kind feature

### What does this PR do / why do we need it:

Fixes the output when data contains '\0'

### Which issue(s) this PR fixes:

Fixes #124  #I3BXGG

### Special notes for your reviewers:

### CODE REVIEW

**Review by Nitin-Kashyap — CHANGES_REQUESTED** [review]

File: presto-rcfile/src/main/java/io/prestosql/rcfile/text/StringEncoding.java [hunk 1]
```diff
@@ -41,21 +42,113 @@ public StringEncoding(Type type, Slice nullSequence, Byte escapeChar)
...
+                }
+                else {
+                    output.writeBytes(slice);
+                }
             }
             encodeOutput.closeEntry();
         }
     }
 
+    private Text unescapeText(Text text)
```
[the remark this answers is not in the document]
Comment (aman-omer) [reply presto-rcfile/src/main/java/io/prestosql/rcfile/text/StringEncoding.java]: Moved function to `presto-rcfile/src/main/java/io/prestosql/rcfile/RcFileDecoderUtils.java`

File: presto-rcfile/src/main/java/io/prestosql/rcfile/binary/StringEncoding.java [hunk 2]
```diff
@@ -43,20 +45,112 @@ public StringEncoding(Type type)
...
+                    else {
+                        output.writeBytes(slice);
+                    }
                 }
             }
             encodeOutput.closeEntry();
         }
     }
 
+    private Text unescapeText(Text text)
```
[the remark this answers is not in the document]
Comment (aman-omer) [reply presto-rcfile/src/main/java/io/prestosql/rcfile/binary/StringEncoding.java]: Moved function to `presto-rcfile/src/main/java/io/prestosql/rcfile/RcFileDecoderUtils.java`

File: presto-hive/src/main/java/io/prestosql/plugin/hive/util/FieldSetterFactory.java [hunk 3]
```diff
@@ -299,7 +299,86 @@ public VarcharFieldSetter(SettableStructObjectInspector rowInspector, Object row
         public void setField(Block block, int position)
         {
             value.set(type.getSlice(block, position).getBytes());
-            rowInspector.setStructFieldData(row, field, value);
+            rowInspector.setStructFieldData(row, field, unescapeText(value));
+        }
+
+        private Text unescapeText(Text text)
```
[the remark this answers is not in the document]
Comment (aman-omer) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/util/FieldSetterFactory.java]: I have used 'org.apache.commons.text.StringEscapeUtils'

File: presto-orc/src/main/java/io/prestosql/orc/writer/DictionaryBuilder.java [hunk 4]
```diff
@@ -174,6 +187,98 @@ private int addNewElement(long hashPosition, Block block, int position)
         return newElementPositionInBlock;
     }
 
+    private int unescapeText(byte[] text)
```
[the remark this answers is not in the document]
Comment (aman-omer) [reply presto-orc/src/main/java/io/prestosql/orc/writer/DictionaryBuilder.java:191]: Done

File: presto-orc/src/main/java/io/prestosql/orc/writer/DictionaryBuilder.java [hunk 5]
```diff
@@ -174,6 +188,30 @@ private int addNewElement(long hashPosition, Block block, int position)
...
+    {
+        String escaped = StringEscapeUtils.unescapeJava(new String(text));
+        byte[] escapedBytes = escaped.getBytes();
```
Comment (Nitin-Kashyap) [inline presto-orc/src/main/java/io/prestosql/orc/writer/DictionaryBuilder.java:194]: use System.arrayCopy etc

Comment (aman-omer) [reply presto-orc/src/main/java/io/prestosql/orc/writer/DictionaryBuilder.java]: Done

**Comment by Nitin-Kashyap** [comment]
lgtm

File: presto-hive/src/main/java/io/prestosql/plugin/hive/GenericHiveRecordCursor.java [hunk 6]
```diff
@@ -377,6 +377,23 @@ else if (fieldValue instanceof HiveCharWritable) {
         }
     }
 
+    private Text escapeText(Text text)
```
[the remark this answers is not in the document]
Comment (aman-omer) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/GenericHiveRecordCursor.java]: Done

**Pull request closed without merging**